### PR TITLE
2 changes for testing in development

### PR DESCRIPTION
**Pull request by BradHN1**
Customizing Registration and Signin
Adding All Actions download for cadmins

### CODE REVIEW

File: app/containers/MassEnergizeSuperAdmin/Community/Profile/About.js [hunk 1]
```diff
@@ -64,8 +64,14 @@ class About extends React.Component {
     let oldLoadingCSVs = this.state.loadingCSVs;
     this.setState({ loadingCSVs: oldLoadingCSVs.concat(endpoint) });
 
-    const csvResponse = await apiCallFile('/downloads.' + endpoint,
-      { community_id: community.id });
+    const body = {};
+    const dot = endpoint.indexOf('.');
+    if (dot > 0) {
```
Comment (sopokuagyemang) [inline app/containers/MassEnergizeSuperAdmin/Community/Profile/About.js:69]: what computation is happening here?  what is `dot` supposed to tell us?  maybe a comment would really help.

**Review by sopokuagyemang — CHANGES_REQUESTED** [review]

File: app/containers/MassEnergizeSuperAdmin/Community/Profile/About.js (hunk 1 above)
Comment (BradHN1) [reply app/containers/MassEnergizeSuperAdmin/Community/Profile/About.js:69]: Clarified with a comment, slight update to code

File: app/containers/App/Application.js [hunk 2]
```diff
@@ -152,6 +153,8 @@ class Application extends React.Component {
           <Route path="/admin/edit/:id/teams" component={TeamsPage} />
           <Route path="/admin/edit/:id/vendors" component={VendorsPage} />
           <Route path="/admin/edit/:id/testimonials" component={TestimonialsPage} />
+          <Route path="/admin/edit/:id/registration" component={RegisterPage} />
```
Comment (sopokuagyemang) [inline app/containers/App/Application.js:156]: i'm surprised this was not here before.  What is the intent behind adding it if it was not here already?

File: app/containers/MassEnergizeSuperAdmin/Community/Profile/About.js [hunk 3]
```diff
@@ -56,16 +56,24 @@ class About extends React.Component {
...
     this.setState({ loadingCSVs: oldLoadingCSVs.concat(endpoint) });
 
-    const csvResponse = await apiCallFile('/downloads.' + endpoint,
-      { community_id: community.id });
+    // csv downloads can be for a particular community, or they can be for all communities in which case the endpoint ends with '.all'.  
+    // In that case need to remove the '.all' from the endpoint to the API and not send the community ID in the body
+    const body = {};
+    const dotAll = endpoint.indexOf('.all');
+    if (dotAll > 0) {
+      endpoint = endpoint.substring(0,dotAll)
```
Comment (sopokuagyemang) [inline app/containers/MassEnergizeSuperAdmin/Community/Profile/About.js:72]: thanks for adding this.  I will not press too much but i think the gymnastics going on here shows that we should be having to separate functions in the first place.  Determining what to call based on an endpoint is quite weird.

**Review by sopokuagyemang — APPROVED** [review]

File: app/containers/App/Application.js (hunk 2 above)
Comment (BradHN1) [reply app/containers/App/Application.js:156]: This call was missing - when I tested the ability to customize the registration page I forgot to commit this file.

**Pull request merged**